### ISSUE TITLE
Oncoplot % of mutations 

### ISSUE THREAD

**Issue by pradeep-sidd** [PoisonAlien/maftools#930]
When I run the oncoplot code for a TCGA cancer dataset with the clinical information I get an oncoplot with the top mutated genes and their percentages (as a part of the total samples) . For example, when I use cholangiocarcinoma dataset, the oncoplot reads as 29 (0.18%) of 16419 samples altered. I understand that this is happening because each patient has multiple mutations/ can, but as a clinician, I want to get an output with the percentages more related to patients.

**Command**
# creating a maf object

cholang.cna.clinical = read.maf(maf = cholang_mutations,
                                cnTable = cholang_cna,
                clinicalData = data_clinical_patient,
                verbose = FALSE)
[Choalangiocarcinoma.zip](https://github.com/PoisonAlien/maftools/files/11291293/Choalangiocarcinoma.zip)

# GETTING THE BASIC ONCOPLOT
oncoplot(maf = cholang.cna.clinical, top = 15)

getClinicalData(x=cholang.cna.clinical)

####Session info
R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22621)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pheatmap_1.0.12                   NMF_0.26                         
 [3] Biobase_2.58.0                    cluster_2.1.4                    
 [5] rngtools_1.5.2                    registry_0.5-1                   
 [7] R.utils_2.12.2                    R.oo_1.25.0                      
 [9] R.methodsS3_1.8.2                 tidyr_1.3.0                      
[11] data.table_1.14.8                 maftools_2.14.0                  
[13] dplyr_1.1.1                       BSgenome.Hsapiens.UCSC.hg19_1.4.3
[15] BSgenome_1.66.3                   rtracklayer_1.58.0               
[17] Biostrings_2.66.0                 XVector_0.38.0                   
[19] GenomicRanges_1.50.2              GenomeInfoDb_1.34.9              
[21] IRanges_2.32.0                    S4Vectors_0.36.2                 
[23] BiocGenerics_0.44.0               mutSignatures_2.1.5              
[25] foreach_1.5.2                    

loaded via a namespace (and not attached):
 [1] bitops_1.0-7                matrixStats_0.63.0          fs_1.6.1                   
 [4] usethis_2.1.6               devtools_2.4.5              RColorBrewer_1.1-3         
 [7] doParallel_1.0.17           tools_4.2.2                 profvis_0.3.7              
[10] utf8_1.2.3                  R6_2.5.1                    colorspace_2.1-0           
[13] DNAcopy_1.72.3              urlchecker_1.0.1            withr_2.5.0                
[16] tidyselect_1.2.0            prettyunits_1.1.1           processx_3.8.0             
[19] compiler_4.2.2              cli_3.6.1                   DelayedArray_0.24.0        
[22] labeling_0.4.2              scales_1.2.1                callr_3.7.3                
[25] proxy_0.4-27                stringr_1.5.0               digest_0.6.31              
[28] Rsamtools_2.14.0            pkgconfig_2.0.3             htmltools_0.5.5            
[31] sessioninfo_1.2.2           MatrixGenerics_1.10.0       fastmap_1.1.1              
[34] htmlwidgets_1.6.2           rlang_1.1.0                 rstudioapi_0.14            
[37] shiny_1.7.4                 farver_2.1.1                BiocIO_1.8.0               
[40] generics_0.1.3              BiocParallel_1.32.6         RCurl_1.98-1.12            
[43] magrittr_2.0.3              GenomeInfoDbData_1.2.9      Matrix_1.5-3               
[46] Rcpp_1.0.10                 munsell_0.5.0               fansi_1.0.4                
[49] lifecycle_1.0.3             stringi_1.7.12              yaml_2.3.7                 
[52] SummarizedExperiment_1.28.0 zlibbioc_1.44.0             plyr_1.8.8                 
[55] pkgbuild_1.4.0              grid_4.2.2                  parallel_4.2.2             
[58] promises_1.2.0.1            crayon_1.5.2                miniUI_0.1.1.1             
[61] lattice_0.20-45             splines_4.2.2               ps_1.7.3                   
[64] pillar_1.9.0                rjson_0.2.21                reshape2_1.4.4             
[67] codetools_0.2-19            pkgload_1.3.2               XML_3.99-0.14              
[70] glue_1.6.2                  BiocManager_1.30.20         remotes_2.4.2              
[73] vctrs_0.6.1                 httpuv_1.6.9                gtable_0.3.3               
[76] purrr_1.0.1                 cachem_1.0.7                ggplot2_3.4.1              
[79] gridBase_0.4-7              mime_0.12                   xtable_1.8-4               
[82] restfulr_0.0.15             pracma_2.4.2                later_1.3.0                
[85] survival_3.5-5              tibble_3.2.1                iterators_1.0.14           
[88] GenomicAlignments_1.34.1    memoise_2.0.1               ellipsis_0.3.2

**Comment by PoisonAlien** [comment]
Hi,

> I want to get an output with the percentages more related to patients.

Could you give an example of the expected output?

**Comment by pradeep-sidd** [comment]
Thank for the reply Anand.
From the output I am getting it seems like every single mutation is being treated as occurring in a different sample, but if you look at the sample ID, there are multiple repetitions.
For example in the plot I have attached as a pdf, the top 20 mutations are altered in 33 patients, which is represented as 0.2% of the 16419 samples. These samples are not individual patients, I would for instance for the top mutation which is BAP1 to be depicted as 21.21%, which is the fraction of the patients affected. I don’t know if that makes sense to you.

Best

Pradeep

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by PoisonAlien** [comment]
Hi Pradeep,
If understand correctly, you'd like the percentage and mutation frequency to be estimated only for the samples in the `oncoplot`?

**Comment by PoisonAlien** [comment]
Hi,

>Yes, I think that would be useful. 

This would be not the right way since the frequency is estimated at the cohort level.

>And if possible also have patients without the mutations also to be included in the plot.

You can achieve this by setting `removeNonMutated  = FALSE`

**Comment by pradeep-sidd** [comment]
Thank you so much Anand for going over it.
I figured out the problem I was having. Although, I made sure the sample IDs were the same, between the clinical features document and the mutation dataset from TCGA, I had actually changed it to patient ID instead. Once I made sure the Tumor_Sample_ID in the mutation dataset was unchanged and the clinical dataset reflected the same, I got the oncoplot the way I wanted.

Best

Pradeep


Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows
